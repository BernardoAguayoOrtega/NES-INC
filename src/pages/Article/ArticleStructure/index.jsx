// import react
import React, { useContext, useEffect, useState } from 'react';
// use params hooks
import { useParams } from 'react-router-dom'
// import context
import { Context } from '../../../utils/Context/index.jsx';
// import disqus
import { DiscussionEmbed } from 'disqus-react';

const ArticleStructure = () => {
  // use state
  const [new$, setNew$] = useState()
  // use context
  const { news } = useContext(Context)
  // use ref hooks
  const { name } = useParams()
  // use effect
  useEffect(() => {
    setNew$(news?.data?.find(item => item?.author === name && item))
  }, [])

  return (
    <article className="post">
      <section className="post__content">
        <div className="post__content--headline">
          <h1>{new$?.title}</h1>
        </div>
        <img src={`http://${new$?.images[0]}` || 'https://images.unsplash.com/photo-1441974231531-c6227db76b6e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1051&q=80CdYg35qC4hY8ByncPpj_6MQ&usqp=CAU '} className="post__content--image"/>
        <div className="post__content--text">
          {new$?.body?.map((paragraph, i) => <p key={i}>{paragraph}</p>)}
        </div>
      </section>
      <DiscussionEmbed
        shortname="news-inc"
        config={
          {
            url: `http://localhost:3000/#/article/${new$?.author}`,
            identifier: new$?._id,
            title: new$?.title,
            language: 'en'
          }
        }
      />
    </article>
  )
};

export default ArticleStructure;
