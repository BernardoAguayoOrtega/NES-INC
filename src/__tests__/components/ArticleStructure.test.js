// import react
import React from 'react';
// import shallow
import { mount } from 'enzyme';
// import toJson
import toJson from 'enzyme-to-json';
// import router
import { HashRouter as Router } from 'react-router-dom';
// import app component
import ArticleStructure from '../../pages/Article/ArticleStructure/index.jsx';
// import context provider
import { ContextProvider } from '../../utils/Context/index.jsx';

describe('<Article ...props/>', () => {
  const wrapper = mount(
    <ContextProvider>
      <Router>
        <ArticleStructure />
      </Router>
    </ContextProvider>,
  );

  it('is the Article rendering?', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('is the headline rendering?', () => {
    expect(wrapper.find('.post__content--headline').exists()).toBe(true)
  });

  wrapper.setProps({ $news: { title: 'random', image: 'random' } })

  it('is the src rendering?', () => {
    expect(wrapper.find('.post__content--image').prop('src')).toBe();
  });

  it('matches the snap shot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
