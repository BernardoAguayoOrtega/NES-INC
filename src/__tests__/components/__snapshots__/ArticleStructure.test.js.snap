// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article ...props/> matches the snap shot 1`] = `
<ContextProvider
  $news={
    Object {
      "image": "random",
      "title": "random",
    }
  }
>
  <HashRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ArticleStructure>
        <article
          className="post"
        >
          <section
            className="post__content"
          >
            <div
              className="post__content--headline"
            >
              <h1 />
            </div>
            <img
              className="post__content--image"
            />
            <div
              className="post__content--text"
            >
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec in dui ultricies, tristique justo eu, ornare nisl. Phasellus accumsan vel justo eu volutpat. Mauris augue ipsum, accumsan non nisl ut, rutrum iaculis enim. Pellentesque risus est, mollis sed porttitor sed, efficitur quis enim. Etiam pulvinar lobortis lectus, nec dapibus ex cursus nec. Integer volutpat rutrum turpis ac viverra.
              </p>
              <br />
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec in dui ultricies, tristique justo eu, ornare nisl. Phasellus accumsan vel justo eu volutpat. Mauris augue ipsum, accumsan non nisl ut, rutrum iaculis enim. Pellentesque risus est, mollis sed porttitor sed, efficitur quis enim. Etiam pulvinar lobortis lectus, nec dapibus ex cursus nec. Integer volutpat rutrum turpis ac viverra.
              </p>
              <br />
              <p>
                Sed eu tincidunt risus. Cras at lectus viverra, dictum erat nec, maximus ipsum. Aenean ante arcu, aliquam ut lacinia malesuada, pulvinar ac mi. Curabitur iaculis eu dolor nec consectetur. Aliquam semper volutpat diam, id faucibus leo aliquam id. Phasellus mattis in leo in bibendum. Ut aliquet velit purus, eu viverra eros aliquam sed. Vestibulum maximus turpis vel eros vestibulum, in pharetra dui vestibulum.
              </p>
            </div>
          </section>
          <DiscussionEmbed
            config={
              Object {
                "identifier": "",
                "language": "zh_TW",
                "title": "",
                "url": "",
              }
            }
            shortname="example"
          >
            <div
              config={
                Object {
                  "identifier": "",
                  "language": "zh_TW",
                  "title": "",
                  "url": "",
                }
              }
              id="disqus_thread"
              shortname="example"
            />
          </DiscussionEmbed>
        </article>
      </ArticleStructure>
    </Router>
  </HashRouter>
</ContextProvider>
`;
